dds/DCPS/XTypes/DynamicDataHolder.h

```diff
@@ -0,0 +1,812 @@
+/*
+ * Distributed under the OpenDDS License.
+ * See: http://www.opendds.org/license.html
+ */
+
+#ifndef OPENDDS_DCPS_XTYPES_DYNAMIC_DATA_HOLDER_H
+#define OPENDDS_DCPS_XTYPES_DYNAMIC_DATA_HOLDER_H
+
+#ifndef OPENDDS_SAFETY_PROFILE
+#ifndef OPENDDS_NO_CONTENT_SUBSCRIPTION_PROFILE
+
+#include <dds/DdsDynamicDataC.h>
+#include <dds/DCPS/FilterEvaluator.h>
+
+OPENDDS_BEGIN_VERSIONED_NAMESPACE_DECL
+
+namespace OpenDDS {
+namespace XTypes {
+
+template <typename T>
+class DynamicDataHolder : public DDS::DynamicData {
```

Comment:
Is there a more descriptive name for this? 
AdaptedDynamicData
DynamicAccess
DynamicView